src/Utility/InputValidator.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Input validation utilities.
+ *
+ * @package Requests
+ * @subpackage Utilities
+ */
+
+namespace WpOrg\Requests\Utility;
+
+use ArrayAccess;
+use CurlHandle;
+use Traversable;
+
+/**
+ * Input validation utilities.
+ *
+ * @package Requests
+ * @subpackage Utilities
+ */
+final class InputValidator {
+
+	/**
+	 * Verify that a received input parameter is of type string or is "stringable".
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_string_or_stringable($input) {
+		return is_string($input) || self::is_stringable_object($input);
+	}
+
+	/**
+	 * Verify whether a received input parameter is usable as an integer array key.
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_numeric_array_key($input) {
```

Comment:
Potential alternative names:
- `is_decimal()`
- `is_array_index()`

Reply:
Neither or these feels right...

* `is_decimal()` - floats are also decimal, but not acceptable.
* `is_array_index` - a non-numeric string array index is also a valid array index, but for the purpose of this function should be ignored (return `false`).